libs/@guardian/react-crossword/src/components/Crossword.tsx

```diff
@@ -15,6 +15,9 @@ import { Clues } from './Clues';
 import { Controls } from './Controls';
 import { Grid } from './Grid';
 
+// define and cache the regex for valid keydown events
+const keyDownRegex = /^[A-Za-zÀ-ÿ0-9]$/;
```

Comment:
do we want lowercase a-z ?

Reply:
that's a good point. if we've already `uppercase`d it there's no need (and it's also what's happening [in frontend](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/common/modules/crosswords/crossword.js#L361)). 

I've slightly rejigged the logic to test against all valid keyboard inputs, and then store a valid input as uppercased if it passes.